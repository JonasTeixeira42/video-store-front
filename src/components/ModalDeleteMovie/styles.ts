import styled, { css } from 'styled-components';

import * as ButtonStyles from 'components/Button/styles';

export const Wrapper = styled.div``;

export const Text = styled.div`
  & > span {
    font-size: 1.8rem;
  }
`;

export const OperationsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: flex-end;
    margin-top: ${theme.spacings.small};

    ${ButtonStyles.Wrapper} {
      margin-left: ${theme.spacings.xxsmall};
    }
  `}
`;
