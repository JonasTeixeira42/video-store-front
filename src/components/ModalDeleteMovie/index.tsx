import React from 'react';

import Modal, { ModalProps } from 'components/Modal';
import { Movie } from 'components/ModalEditMovie';
import Button from 'components/Button';

import * as S from './styles';

export type ModalDeleteMoviesProps = {
  movie: Movie;
  onDelete: (id: string) => void;
} & Omit<ModalProps, 'children'>;

const ModalDeleteMovie = ({
  movie,
  isOpen = false,
  onDelete,
  setIsOpen,
}: ModalDeleteMoviesProps) => {
  const handleSubmitDelete = () => {
    !!onDelete && onDelete(movie.id);
  };

  return (
    <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
      <S.Wrapper>
        <S.Text>
          <span>{`Deseja remover o filme ${movie.name}?`}</span>
        </S.Text>
        <S.OperationsWrapper>
          <Button size="small" background="secondary" onClick={setIsOpen}>
            NÃ£o
          </Button>
          <Button
            size="small"
            background="primary"
            onClick={() => onDelete(movie.id)}
          >
            Sim
          </Button>
        </S.OperationsWrapper>
      </S.Wrapper>
    </Modal>
  );
};

export default ModalDeleteMovie;
