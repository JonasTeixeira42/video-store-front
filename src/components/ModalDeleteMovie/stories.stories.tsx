import { Story, Meta } from '@storybook/react/types-6-0';

import ModalDeleteMovie, { ModalDeleteMoviesProps } from '.';

export default {
  title: 'ModalDeleteMovie',
  component: ModalDeleteMovie,
  args: {
    movie: {
      id: '1',
      name: 'Star Wars',
    },
    isOpen: true,
  },
} as Meta;

export const Default: Story<ModalDeleteMoviesProps> = (args) => (
  <ModalDeleteMovie {...args} />
);
