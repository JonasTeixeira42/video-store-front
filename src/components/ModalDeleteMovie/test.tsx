import { screen, fireEvent } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import ModalDeleteMovie from '.';

const mock = {
  id: '1',
  name: 'Ghost in the shell',
  image: 'https://source.unsplash.com/300x140',
  url: 'https://source.unsplash.com/300x140',
  director: {
    id: '1',
    name: 'Masamune Shirow',
  },
};

describe('<ModalDeleteMovie />', () => {
  it('should render the Modal correctly', () => {
    const { container } = renderWithTheme(
      <ModalDeleteMovie movie={mock} isOpen={true} />,
    );

    expect(screen.getByText(/ghost in the shell/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /sim/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /nÃ£o/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onDelete function when passed', () => {
    const onDelete = jest.fn();

    renderWithTheme(
      <ModalDeleteMovie movie={mock} isOpen={true} onDelete={onDelete} />,
    );

    const button = screen.getByRole('button', { name: /sim/i });

    fireEvent.click(button);

    expect(onDelete).toHaveBeenCalled();
    expect(onDelete).toHaveBeenLastCalledWith(mock.id);
  });
});
