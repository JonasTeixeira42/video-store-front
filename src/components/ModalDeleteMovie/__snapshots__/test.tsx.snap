// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalDeleteMovie /> should render the Modal correctly 1`] = `null`;
